packages/field-label/src/FieldLabel.ts

```diff
@@ -89,25 +92,33 @@ export class FieldLabel extends SizedMixin(SpectrumElement) {
 
     private resolvedElement = new ElementResolutionController(this);
 
-    private addTarget(target: Focusable): void {
-        this.target = target.focusElement || target;
-        const targetParent = this.target.getRootNode() as HTMLElement;
-        if (targetParent === (this.getRootNode() as HTMLElement)) {
-            conditionAttributeWithId(this.target, 'aria-labelledby', [this.id]);
+    private addTarget(target: Labelable): void {
+        this.target = target;
+        const applyLabel = this.target.applyFocusElementLabel;
+        const focusable = this.target.focusElement || this.target;
+        const targetParent = focusable.getRootNode() as HTMLElement;
+        if (typeof applyLabel !== 'undefined') {
+            applyLabel(this.labelText);
+        } else if (targetParent === (this.getRootNode() as HTMLElement)) {
+            conditionAttributeWithId(focusable, 'aria-labelledby', [this.id]);
         } else {
-            this.target.setAttribute('aria-label', this.labelText);
+            focusable.setAttribute('aria-label', this.labelText);
         }
     }
 
     private removeTarget(): void {
         if (this.target) {
-            const targetParent = this.target.getRootNode() as HTMLElement;
-            if (targetParent === (this.getRootNode() as HTMLElement)) {
-                conditionAttributeWithoutId(this.target, 'aria-labelledby', [
+            const applyLabel = this.target.applyFocusElementLabel;
+            const focusable = this.target.focusElement || this.target;
+            const targetParent = focusable.getRootNode() as HTMLElement;
+            if (typeof applyLabel !== 'undefined') {
+                applyLabel();
+            } else if (targetParent === (this.getRootNode() as HTMLElement)) {
+                conditionAttributeWithoutId(focusable, 'aria-labelledby', [
                     this.id,
                 ]);
             } else {
-                this.target.removeAttribute('aria-label');
+                focusable.removeAttribute('aria-label');
             }
```

Comment:
Is this identical (minus `this.labelText -> undefined`) to `addTarget`? If so, you could extract this to something like `setTarget(value: string)`, then `addTarget` could call `setTarget(this.labelText)` and `removeTarget` could call `setTarget(undefined)`. Unnecessary, just something to consider.